super_calendar/models/super_calendar_configurator.py

```diff
@@ -0,0 +1,160 @@
+        # Remove old records
+        self._clear_super_calendar_records()
+
+        # Rebuild all calendar records
+        configurator_list = self.search([])
+        for configurator in configurator_list:
+            for line in configurator.line_ids:
+                configurator._generate_record_from_line(line)
+        _logger.info('Calendar generated')
+        return True
+
+    @api.multi
+    def _generate_record_from_line(self, line):
+        """
+        Create super_calendar records from super_calendar_configurator_line
+        objects.
+        """
+        super_calendar_pool = self.env['super.calendar']
+        values = self._get_record_values_from_line(line)
+        for record in values:
+            super_calendar_pool.create(values[record])
+
+    @api.multi
+    def _get_record_values_from_line(self, line):
+        """
+        Get super_calendar fields values from super_calendar_configurator_line
+        objects.
+        Check if the User value is a res.users.
+        """
+        res = {}
+        current_pool = self.env[line.name.model]
+        domain = line.domain and safe_eval(line.domain) or []
+        current_record_list = current_pool.search(domain)
+        for cur_rec in current_record_list:
+            f_user = line.user_field_id.name
+            f_descr = line.description_field_id.name
+            f_date_start = line.date_start_field_id.name
+            f_date_stop = line.date_stop_field_id.name
+            f_duration = line.duration_field_id.name
+            if (f_user and
+                    cur_rec[f_user] and
+                    cur_rec[f_user]._model._name != 'res.users'):
+                raise Exception(
+                    _('Error'),
+                    _("The 'User' field of record %s (%s) "
+                      "does not refer to res.users")
+                    % (cur_rec[f_descr], line.name.model))
+
+            if (((f_descr and cur_rec[f_descr]) or
+                    line.description_code) and
+                    cur_rec[f_date_start]):
+                duration = False
+                if (not line.duration_field_id and
+                        line.date_stop_field_id and
+                        cur_rec[f_date_start] and
+                        cur_rec[f_date_stop]):
+                    date_start_str = cur_rec[f_date_start]
+                    date_stop_str = cur_rec[f_date_stop]
+                    if line.date_start_field_id.ttype == 'date':
+                        date_start_str += ' 12:00:00'
+                    if line.date_stop_field_id.ttype == 'date':
+                        date_stop_str += ' 12:00:00'
+                    date_start = datetime.strptime(
+                        date_start_str,
+                        tools.DEFAULT_SERVER_DATETIME_FORMAT
+                    )
+                    date_stop = datetime.strptime(
+                        date_stop_str,
+                        tools.DEFAULT_SERVER_DATETIME_FORMAT
+                    )
```

Comment:
The way the time is added to the string should be done using datetime utils

Currently it is:
date string -> datetime string with 12:00:00 as time -> datetime.date object -> datetime.datetime object
It should be:
date string -> datetime.date object -> datetime.datetime object with 12:00:00 as time